lib/flow/VaporField.cpp

```diff
@@ -0,0 +1,424 @@
+        if( time == _timestamps[floorTS] )
+            velocity = floorVelocity * mult; 
+        else
+        {
+            for( int i = 0; i < 3; i++ )
+            {
+                auto varname = VelocityNames[i];
+                int     rv   = _getAGrid( floorTS + 1, varname, &grid );
+                assert( rv  == 0 );
+                ceilVelocity[i] = grid->GetValue( coords );
+                missingV[i]     = grid->GetMissingValue();
+            }
+            hasMissing = glm::equal( ceilVelocity, missingV );
+            if( glm::any( hasMissing ) )
+            {
+                velocity = glm::vec3( 0.0f );
+                return 0;
+            }
+            
+            float weight = (time - _timestamps[floorTS]) / 
+                           (_timestamps[floorTS+1] - _timestamps[floorTS]);
+            velocity = glm::mix( floorVelocity, ceilVelocity, weight ) * mult;
+        }
+    }
+
+    return 0;
+}
+
+
+int
+VaporField::GetScalar( float time, const glm::vec3& pos, float& scalar,
+                       bool  checkInsideVolume )
+{
+    if( ScalarName.empty() )
+        return NO_FIELD_YET;
+    if( checkInsideVolume )
+        if( !InsideVolumeScalar( time, pos ) )
+            return OUT_OF_FIELD;
+
+    std::string scalarname = ScalarName;    // const requirement...
+
+    const std::vector<double> coords{ pos.x, pos.y, pos.z };
+    const VAPoR::Grid* grid = nullptr;
+
+    if( IsSteady )
+    {
+        size_t currentTS = _params->GetCurrentTimestep();
+        int     rv = _getAGrid( currentTS, scalarname, &grid );
+        assert( rv == 0 );
+        float gridV = grid->GetValue( coords );
+        if( gridV  == grid->GetMissingValue() )
+            scalar  = 0.0f;
+        else
+            scalar  = gridV;
+    }
+    else
+    {
+        // First check if the query time is within range
+        if( time < _timestamps.front() || time > _timestamps.back() )
+            return TIME_ERROR;
+
+        // Then we locate the floor time step
+        size_t floorTS;
+        int rv  = LocateTimestamp( time, floorTS );
+        assert( rv == 0 );
+        rv          = _getAGrid( floorTS, scalarname, &grid );
+        assert( rv == 0 );
+        float floorScalar = grid->GetValue( coords );
+        if( floorScalar  == grid->GetMissingValue() )
+        {
+            scalar = 0.0f;
+            return 0;
+        }
+
+        if( time == _timestamps[floorTS] )
+            scalar = floorScalar;
+        else
+        {
+            rv = _getAGrid( floorTS + 1, scalarname, &grid );
+            assert( rv == 0 );
+            float ceilScalar = grid->GetValue( coords );
+            if( ceilScalar  == grid->GetMissingValue() )
+            {
+                scalar = 0.0f;
+                return 0;
+            }
+            float weight = (time - _timestamps[floorTS]) /
+                           (_timestamps[floorTS+1] - _timestamps[floorTS]);
```

Comment:
Check for divide by zero would be good.

Reply:
It's really an abnormal case to have zero here, so I added a `VAssert()` a few lines above.